File: LeetCode/392.is-subsequence.py
#
# @lc app=leetcode id=392 lang=python3
#
# [392] Is Subsequence
#


# @lc code=start
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        index_t = 0
        for char_s in s:
            while index_t < len(t) and char_s != t[index_t]:
                index_t += 1
            if index_t >= len(t):
                return False
            index_t += 1

        return True


# @lc code=end

File: LeetCode/392.is-subsequence.cpp
/*
 * @lc app=leetcode id=392 lang=cpp
 *
 * [392] Is Subsequence
 */

// @lc code=start
class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        int tLen = t.length(), sLen = s.length();
        if (tLen < sLen)
            return false;
        for (int i = 0, j = 0; i < tLen || j == sLen; i++)
        {
            if (t[i] == s[j])
            {
                j++;
                continue;
            }
            if (sLen - j >= tLen - i)
            {
                return false;
            }
        }
        return true;
    }
};
// @lc code=end
